source/adaptive_load/metrics_plugin_impl.cc

```diff
@@ -0,0 +1,123 @@
+#include "adaptive_load/metrics_plugin_impl.h"
+
+#include <cmath>
+
+#include "envoy/registry/registry.h"
+
+#include "external/envoy/source/common/common/assert.h"
+#include "external/envoy/source/common/protobuf/protobuf.h"
+
+#include "absl/status/status.h"
+#include "absl/strings/numbers.h"
+
+namespace Nighthawk {
+
+NighthawkStatsEmulatedMetricsPlugin::NighthawkStatsEmulatedMetricsPlugin(
+    const nighthawk::client::Output& nighthawk_output) {
+  bool found_global_result = false;
+  for (const nighthawk::client::Result& result : nighthawk_output.results()) {
+    if (result.name() != "global") {
```

Comment:
To the above comment - e.g. this portion could be a helper function that retrieves the global result. And so on.

Reply:
Done.